CONTRIBUTING.md

```diff
@@ -0,0 +1,216 @@
+
+### Make a local copy of GitHub repository
+
+> **Supported Python environments**
+>We strongly recommend the [Anaconda](https://www.anaconda.com/distribution/) or [Miniconda](https://conda.io/en/latest/miniconda.html) environment managers for Python.
+
+You must clone the Nerosim-lab/netpyne repository from your remote fork, and also connect the local copy to the `upstream` version of the codebase, so you can stay up-to-date with changes from other contributors. First, edit these two variables for your situation:
+
+Then make a local clone of your remote fork (origin):
+
+```bash
+ ~$ cd $INSTALL_LOCATION
+ ~$ git clone https://github.com/$GITHUB_USERNAME/netpyne.git
+```
+
+> **Remote URLs in git**
+>Here we use `git://` instead of `https://` in the URL for the `upstream` remote repository. `git://` URLs are read-only, so you can `pull` changes from `upstream` into your local copy (to stay up-to-date with changes from other contributors) but you cannot `push` changes from your computer into the `upstream` remote. Instead, you must `push` your changes to your own remote fork (`origin`) first, and then create a pull request from your remote into the `upstream` remote. In [a later section](#connecting-to-github-with-ssh-optional) you’ll see a third kind of remote URL for connecting to GitHub using SSH.
+
+Finally, set up a link between your local clone and the official repository (upstream):
+
+```bash
+~$ cd netpyne
+~$ git remote add upstream https://github.com/Neurosim-lab/netpyne.git
+~$ git fetch --all
+~$ git remove -v   #Shows your tracked repositories.
+```
+
+### Basic git commands
+
+Learning to work with git can take a long time, because it is a complex and powerful tool for managing versions of files across multiple users, each of whom have multiple copies of the codebase. We’ve already seen in the setup commands above a few of the basic git commands useful:
+
+* `git clone <URL_OF_REMOTE_REPO>` (make a local copy of a repository).
+* `git remote add <NICKNAME_OF_REMOTE> <URL_OF_REMOTE_REPO>` (connect a local copy to an additional remote).
+* `git fetch --all` (get the current state of connected remote repos).
+
+Other commands that you will undoubtedly need relate to [branches](https://help.github.com/en/articles/about-branches). Branches represent multiple copies of the codebase within a local clone or remote repo. Branches are typically used to experiment with new features while still keeping a clean, working copy of the original codebase that you can switch back to at any time. The default branch of any repo is always called `master`, and it is recommended that you reserve the `master` branch to be that clean copy of the working `upstream` codebase. 

```

Comment:
`related to branches can be found here` maybe

Reply:
:+1: